source/MaterialXTest/MaterialXCore/Units.cpp

```diff
@@ -191,32 +189,32 @@ TEST_CASE("UnitDocument", "[units]")
 
                 if (pNode->getParameterCount()) {
                     for (mx::ParameterPtr param: pNode->getParameters()) {
-                        const mx::TypeDesc* type = mx::TypeDesc::get(param->getType());
+                        const std::string type = param->getType();
                         const mx::ValuePtr value = param->getValue();
                         if (param->hasUnit() && value) {
 
-                            if (type->isScalar() && type->getBaseType() == mx::TypeDesc::BASETYPE_FLOAT)
+                            if (type == "float")
                             {
                                 float originalval = value->asA<float>();
                                 float convertedValue = uconverter->convert(originalval, param->getUnit(), DISTANCE_DEFAULT);
                                 float reconvert = uconverter->convert(convertedValue, DISTANCE_DEFAULT, param->getUnit());
                                 REQUIRE((originalval - reconvert) < EPSILON);
                             }
-                            else if (type->isFloat2())
+                            else if (type == "vector2" || type == "color2")
```

Comment:
Same comment about not allowing color types applies here.

Reply:
Fixed